src/test/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateContractPartyCsvServiceImplTest.java

```diff
@@ -158,34 +103,179 @@ void testGetJaggerContractsDto() {
+        JaggaerContractAddressUploadDto dto = VendorAddressFixture.FULL_VENDOR_FULL_ADDRESS.toJaggaerContractAddressUploadDto();
+        dto.setNotes(notes);
+        return dto;
+    }
+
+    static String[] buildJaggaerContractAddressUploadDtoData(String notes) {
+        VendorAddressFixture address = VendorAddressFixture.FULL_VENDOR_FULL_ADDRESS;
+        String[] record = { address.rowType.name(), address.addressID,
+                address.sciQuestID, address.name, address.addressType.jaggaerAddressType, address.primaryType, 
+                address.active, address.country, address.streetLine1,
+                address.streetLine2, address.streetLine3, address.city, address.state,
+                address.postalCode, address.phone, address.tollFreeNumber, address.fax,
+                notes };
+        return record;
+    }
+    
+    @Test
+    void testBuildPartyHeader() {
+        String expectedPartyHeader = "PARTY,OverrideDupError,ERPNumber,SciQuestID,ContractPartyName,DoingBusinessAs,OtherNames,CountryOfOrigin,Active,ContractPartyType,Primary,LegalStructure,TaxIDType,TaxIdentificationNumber,VATRegistrationNumber,WebsiteURL";
+        String actualPartyHeader = printArrayCommaDelim(jaggaerGenerateContractPartyCsvServiceImpl.buildPartyHeader());
+        assertEquals(expectedPartyHeader, actualPartyHeader);
+    }
+
+    @Test
+    void testBuildAddressHeader() {
+        String expectedAddressHeader = "ADDRESS,AddressID,SciQuestID,Name,AddressType,PrimaryType,Active,Country,StreetLine1,StreetLine2,StreetLine3,City/Town,State/Province,PostalCode,Phone,TollFreeNumber,Fax,Notes";
+        String actualAddressHeader = printArrayCommaDelim(jaggaerGenerateContractPartyCsvServiceImpl.buildAddressHeader());
+        assertEquals(expectedAddressHeader, actualAddressHeader);
+    }
+
+    @Test
+    void testBuildContactHeader() {
+        String expectedContactHeader = "CONTACT,ContactID,SciQuestID,Name,FirstName,LastName,ContactType,PrimaryType,Active,Title,Email,Phone,MobilePhone,TollFreeNumber,Fax,Notes";
+        String actualContactHeader = printArrayCommaDelim(jaggaerGenerateContractPartyCsvServiceImpl.buildContactHeader());
+        assertEquals(expectedContactHeader, actualContactHeader);
+    }
+
+    private String printArrayCommaDelim(String[] data) {
+        return StringUtils.join(data, KFSConstants.COMMA);
+    }
+    
+    @Test
+    void testGenerateCsvFile() throws IOException {
+        List<JaggaerContractUploadBaseDto> jaggaerUploadDtos = new ArrayList<>();
+        jaggaerUploadDtos.add(VendorFixture.FULL_VENDOR_FOR_CSV.toJaggaerContractPartyUploadDto());
+        jaggaerUploadDtos.add(VendorAddressFixture.FULL_VENDOR_FULL_ADDRESS_FOR_CSV.toJaggaerContractAddressUploadDto());
+        
+        createTargetTestDirectory();
+        
+        jaggaerGenerateContractPartyCsvServiceImpl.generateCsvFile(jaggaerUploadDtos, JaggaerContractUploadProcessingMode.PO);
```

Comment:
Should this call be placed inside the method's "try" block?

Reply:
I will move it